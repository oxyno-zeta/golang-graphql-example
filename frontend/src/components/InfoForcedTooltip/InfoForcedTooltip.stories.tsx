import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import InfoForcedTooltip, { Props } from './InfoForcedTooltip';

export default {
  title: 'Components/InfoForcedTooltip',
  component: InfoForcedTooltip,
  args: { tooltipTitle: 'Fake tooltip !' },
} as Meta<typeof InfoForcedTooltip>;

export const Playground: StoryFn<typeof InfoForcedTooltip> = function C(args: Props) {
  return <InfoForcedTooltip {...args} />;
};
