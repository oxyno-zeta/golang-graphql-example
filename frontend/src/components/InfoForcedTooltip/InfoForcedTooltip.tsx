import React, { ReactNode } from 'react';
import SvgIcon, { SvgIconProps } from '@mui/material/SvgIcon';
import { mdiInformation } from '@mdi/js';
import ForcedTooltip, { ForcedTooltipProps } from '../ForcedTooltip';

export interface Props {
  tooltipTitle: ReactNode;
  tooltipProps?: Omit<ForcedTooltipProps, 'title'>;
  svgIconProps?: Omit<SvgIconProps, 'onClick'>;
  svgIconContainerStyle?: Record<string, string>;
}

const defaultProps = {
  tooltipProps: {},
  svgIconProps: {},
  svgIconContainerStyle: { padding: '8px', verticalAlign: 'middle', textAlign: 'center', display: 'inline-flex' },
};

function InfoForcedTooltip({ tooltipTitle, tooltipProps, svgIconProps, svgIconContainerStyle }: Props) {
  return (
    <ForcedTooltip
      title={<>{tooltipTitle}</>}
      {...tooltipProps}
      render={(handleTooltipOpen) => (
        <span style={svgIconContainerStyle}>
          <SvgIcon onClick={() => handleTooltipOpen()} {...svgIconProps}>
            <path d={mdiInformation} />
          </SvgIcon>
        </span>
      )}
    />
  );
}

InfoForcedTooltip.defaultProps = defaultProps;

export default InfoForcedTooltip;
