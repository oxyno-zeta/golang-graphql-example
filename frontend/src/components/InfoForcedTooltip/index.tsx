import InfoForcedTooltip, { type Props as InfoForcedTooltipProps } from './InfoForcedTooltip';

export default InfoForcedTooltip;
export type { InfoForcedTooltipProps };
