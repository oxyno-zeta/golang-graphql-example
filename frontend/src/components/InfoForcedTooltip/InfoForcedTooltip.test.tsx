import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import InfoForcedTooltip from './InfoForcedTooltip';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('InfoForcedTooltip', () => {
  it('should display text tooltip on mouse click', async () => {
    const { container, getByText, findByRole } = render(<InfoForcedTooltip tooltipTitle="fake-tooltip" />);

    expect(container).toMatchSnapshot();

    // Get child
    const child = container.querySelector('svg');

    expect(child).not.toBeNull();

    expect(fireEvent.click(child as SVGElement)).toBeTruthy();
    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);
    await waitFor(() => {
      expect(getByText('fake-tooltip')).toBeInTheDocument();
    });
    expect(findByRole('tooltip')).not.toBeNull();

    expect(container).toMatchSnapshot();
    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);
  });

  it('should display text tooltip on mouse over only', async () => {
    const { container, getByText, findByRole } = render(<InfoForcedTooltip tooltipTitle="fake-tooltip" />);

    expect(container).toMatchSnapshot();

    // Get child
    const child = container.querySelector('span');

    expect(child).not.toBeNull();

    expect(fireEvent.mouseOver(child as HTMLSpanElement)).toBeTruthy();
    await waitFor(() => {
      expect(getByText('fake-tooltip')).toBeInTheDocument();
    });
    expect(findByRole('tooltip')).not.toBeNull();

    expect(container).toMatchSnapshot();

    expect(fireEvent.mouseLeave(child as HTMLSpanElement)).toBeTruthy();
    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);
    await waitFor(() => {
      expect(container).not.toHaveTextContent('fake-tooltip');
    });

    expect(container).toMatchSnapshot();
  });

  it('should display element tooltip on mouse over only', async () => {
    const { container, getByText, findByRole } = render(<InfoForcedTooltip tooltipTitle={<>fake-tooltip</>} />);

    expect(container).toMatchSnapshot();

    // Get child
    const child = container.querySelector('span');

    expect(child).not.toBeNull();

    expect(fireEvent.mouseOver(child as HTMLSpanElement)).toBeTruthy();
    await waitFor(() => {
      expect(getByText('fake-tooltip')).toBeInTheDocument();
    });
    expect(findByRole('tooltip')).not.toBeNull();

    expect(container).toMatchSnapshot();

    expect(fireEvent.mouseLeave(child as HTMLSpanElement)).toBeTruthy();
    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);
    await waitFor(() => {
      expect(container).not.toHaveTextContent('fake-tooltip');
    });

    expect(container).toMatchSnapshot();
  });

  it('should display element tooltip on mouse click', async () => {
    const { container, getByText, findByRole } = render(<InfoForcedTooltip tooltipTitle={<>fake-tooltip</>} />);

    expect(container).toMatchSnapshot();

    // Get child
    const child = container.querySelector('svg');

    expect(child).not.toBeNull();

    expect(fireEvent.click(child as SVGElement)).toBeTruthy();
    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);
    await waitFor(() => {
      expect(getByText('fake-tooltip')).toBeInTheDocument();
    });
    expect(findByRole('tooltip')).not.toBeNull();

    expect(container).toMatchSnapshot();
    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);
  });
});
