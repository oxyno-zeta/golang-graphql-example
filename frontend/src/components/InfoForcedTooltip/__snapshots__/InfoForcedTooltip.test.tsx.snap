// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoForcedTooltip should display element tooltip on mouse click 1`] = `
<div>
  <div>
    <span
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display element tooltip on mouse click 2`] = `
<div>
  <div>
    <span
      aria-labelledby=":r3:"
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
    <div
      class="MuiTooltip-popper MuiTooltip-popperInteractive css-yem0l2-MuiPopper-root-MuiTooltip-popper MuiPopper-root"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      id=":r3:"
      role="tooltip"
      style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
    >
      <div
        class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom css-ja5taz-MuiTooltip-tooltip"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        fake-tooltip
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display element tooltip on mouse over only 1`] = `
<div>
  <div>
    <span
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display element tooltip on mouse over only 2`] = `
<div>
  <div>
    <span
      aria-labelledby=":r2:"
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
    <div
      class="MuiTooltip-popper MuiTooltip-popperInteractive css-yem0l2-MuiPopper-root-MuiTooltip-popper MuiPopper-root"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      id=":r2:"
      role="tooltip"
      style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
    >
      <div
        class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom css-ja5taz-MuiTooltip-tooltip"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        fake-tooltip
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display element tooltip on mouse over only 3`] = `
<div>
  <div>
    <span
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display text tooltip on mouse click 1`] = `
<div>
  <div>
    <span
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display text tooltip on mouse over only 1`] = `
<div>
  <div>
    <span
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display text tooltip on mouse over only 2`] = `
<div>
  <div>
    <span
      aria-labelledby=":r1:"
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
    <div
      class="MuiTooltip-popper MuiTooltip-popperInteractive css-yem0l2-MuiPopper-root-MuiTooltip-popper MuiPopper-root"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      id=":r1:"
      role="tooltip"
      style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
    >
      <div
        class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom css-ja5taz-MuiTooltip-tooltip"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        fake-tooltip
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoForcedTooltip should display text tooltip on mouse over only 3`] = `
<div>
  <div>
    <span
      class=""
      data-mui-internal-clone-element="true"
      style="padding: 8px; vertical-align: middle; text-align: center; display: inline-flex;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </span>
  </div>
</div>
`;
